tests/unit/test_config.py

```diff
@@ -109,3 +121,136 @@ def test_profile_config_option(
     else:
         assert "spark.kubernetes.executor.request.cores" not in spark_properties
         assert "spark.kubernetes.driver.request.cores" not in spark_properties
+
+
+@patch("managers.auth.AuthenticationManager.set_password")
+@patch("managers.auth.AuthenticationManager.user_exists", return_value=True)
+@patch("managers.k8s.K8sManager.is_namespace_valid", return_value=True)
+@patch("managers.k8s.K8sManager.is_service_account_valid", return_value=True)
+@patch(
+    "events.provider.KyuubiClientProviderEvents.update_clients_endpoints",
+    return_value=True,
+)
+@patch(
+    "managers.service.ServiceManager.get_service_endpoint",
+    return_value="10.10.10.10:10009",
+)
+@patch(
+    "managers.service.ServiceManager.reconcile_services",
+    return_value=True,
+)
+@patch("config.spark.SparkConfig._get_spark_master", return_value="k8s://https://spark.master")
+@patch("managers.k8s.K8sManager.is_s3_configured", return_value=True)
+@patch(
+    "config.spark.SparkConfig._sa_conf", return_value={"spark.hadoop.fs.s3a.endpoint": "foo.bar"}
+)
+@pytest.mark.parametrize(
+    "k8s_selectors",
+    [
+        ("a:b", [("spark.kubernetes.node.selector.a", "b")]),
+        (
+            "a:b,c:d",
```

Comment:
Can we also add more examples of the values with spaces, and symbols in the values (as is validated by the regex)?

some ideas:

```
   foo : bar  , grok : bar
foo : bar-grok , foo1 : bar1-grok1 
```